pkg/controller/azvolume.go

```diff
@@ -65,6 +69,49 @@ type reconcileAzVolume struct {
 var _ reconcile.Reconciler = &reconcileAzVolume{}
 
 func (r *reconcileAzVolume) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
```

Comment:
Rather than have common reconciler to handle both `PV` and `AzVolume` objects, we should have two separate reconciler functions to handle each separately.

Reply:
I don't think controller-runtime pkg gives us the flexibility to set multiple reconcilers for a single controller (https://github.com/kubernetes-sigs/controller-runtime/blob/v0.8.3/pkg/controller/controller.go#L61). We could adopt a custom event handler for PV that appends workload to our separate workqueue but at that point, I think it would be better to create a new dedicated controller to watch PV object.


This ambiguous reconciler approach came from trying to force one controller to reconcile for multiple objects. Controller-runtime docs, too, explicitly states that a single controller should only reconcile for a single object ((https://github.com/kubernetes-sigs/controller-runtime/blob/master/FAQ.md#q-how-do-i-know-which-type-of-object-a-controller-references). But I wasn't sure if adding more controllers to our plugin was desirable.

Reply:
It makes sense to me to create a separate controller rather than try to handle different objects in the same controller. The controller runtime is not optimized for this case.

Reply:
Sounds good 👍